### DISCUSSION TITLE
HoloLens2 example on the workshop

### DISCUSSION THREAD

**Discussion by steveice** [microsoft/psi#237] [Q&A]
Can you provide the HL2 example that demoing on the PSI workshop?

**Comment by sandrist** [comment]
An app with a few simple samples can be found here: https://github.com/microsoft/psi-samples/tree/main/Samples/HoloLensSample

And a collection of customizable apps for capturing, remoting, persisting, and exporting the various HL2 streams can be found here: https://github.com/microsoft/psi/tree/master/Sources/MixedReality/HoloLensCapture

**Reply by steveice** [reply]
Thank you for responding. 
I was looking for the sample code that can run object detection with ONIX model locally and send back to HL2.

**Reply by sandrist** [reply]
We don't have specific sample code released for such an app. I would suggest starting from the HoloLens capture and server apps, and modifying the server app (which runs on the desktop) to include an [ONNX component](https://github.com/microsoft/psi/wiki/ONNX-Overview), and pipe the HoloLens image stream to it. I believe in past examples we were using [Tiny YOLOv2](https://github.com/onnx/models/tree/master/vision/object_detection_segmentation/tiny-yolov2) for the underlying model.

Please don't hesitate to create an issue and ask questions if you start down this path and get stuck.
